src/com/google/enterprise/adaptor/opentext/OpentextAdaptor.java

```diff
@@ -418,6 +431,61 @@ public void init(AdaptorContext context) {
       }
     }
 
+    // This URL is used for search and download.
+    this.indexingContentServerUrl =
+        config.getValue("opentext.indexing.contentServerUrl");
+    if (this.indexingContentServerUrl.isEmpty()) {
+      this.indexingContentServerUrl = this.contentServerUrl;
+    }
+    log.log(Level.CONFIG, "opentext.indexing.contentServerUrl: {0}",
+        this.indexingContentServerUrl);
+
+    try {
+      this.readTimeout = Integer.parseInt(
+          config.getValue("adaptor.docContentTimeoutSecs")) * 1000;
+    } catch (NumberFormatException e) {
+      throw new InvalidConfigurationException(
+          "Invalid value for adaptor.docContentTimeoutSecs: "
+          + config.getValue("adaptor.docContentTimeoutSecs"));
+    }
+    try {
+      this.connectTimeout = Integer.parseInt(
+          config.getValue("adaptor.docHeaderTimeoutSecs")) * 1000;
+    } catch (NumberFormatException e) {
+      throw new InvalidConfigurationException(
+          "Invalid value for adaptor.docHeaderTimeoutSecs: "
+          + config.getValue("adaptor.docHeaderTimeoutSecs"));
+    }
+
+    try {
+      this.downloadMethod = Enum.valueOf(DownloadMethod.class,
+          config.getValue("opentext.indexing.downloadMethod")
+          .toUpperCase(Locale.US));
+    } catch (IllegalArgumentException e) {
+      throw new InvalidConfigurationException(
+          "Invalid value '"
+          + config.getValue("opentext.indexing.downloadMethod")
+          + "' for opentext.indexing.downloadMethod; valid values are "
+          + Arrays.toString(DownloadMethod.values()).toLowerCase(Locale.US));
+    }
+    // Don't log these parameters unless they're set to something
+    // other than the default.
+    if (this.downloadMethod != DownloadMethod.CONTENTSERVER) {
+      log.log(Level.CONFIG, "opentext.indexing.downloadMethod: {0}",
+          this.downloadMethod.name().toLowerCase(Locale.US));
+      if (this.downloadMethod == DownloadMethod.CONTENTHANDLER) {
+        this.downloadHandlerUrl =
```

Comment:
What was your thinking on not pairing these two names (contenthandler and downloadHandlerUrl)?